src/compositions/Timeline/Timeline.tsx

```diff
@@ -1,42 +1,220 @@
+            {
+                id: 'cues',
+                actions: cues.map((cue, index) => ({
+                    id: cue.id,
+                    start: cue.startTime,
+                    end: cue.endTime,
+                    effectId: 'caption',
+                    selected: activeCueIndex === index,
+                })),
+            },
+            {
+                id: 'video',
+                actions: [
+                    {
+                        id: 'video',
+                        start: 0,
+                        end: videoLength,
+                        effectId: 'video',
+                        movable: false,
+                        flexible: false,
+                    },
+                ],
+                rowHeight: 80,
+            },
+        ])
+    }, [cues, activeCueIndex])
+
+    return (
+        <div
+            className={`timeline ${className}`}
+            style={{ '--current-time': currentTime } as CSSProperties}
+        >
+            <div className="timeline__controls">
+                <button
+                    onClick={increaseZoom}
+                    className="timeline__controls__button button button--icon"
+                >
+                    <Icon name="add" />
+                </button>
+                <button
+                    onClick={decreaseZoom}
+                    className="timeline__controls__button button button--icon"
                 >
-                    <button onClick={() => onCueClick(index)} className="timeline__cue-button">
-                        <span className="timeline__cue-id">{cue.id}</span>
-                    </button>
-                </li>
-            ))}
-        </ul>
-
-        <div className="timeline__playhead"></div>
-    </div>
-)
+                    <Icon name="subtract" />
+                </button>
+            </div>
+
+            <ReactTimeline
+                ref={timelineState}
+                editorData={timelineData}
+                effects={timelineEffects}
+                rowHeight={50}
+                autoScroll={true}
+                dragLine={true}
+                scale={Math.floor(scale)}
+                scaleWidth={scaleWidth}
+                getActionRender={
+                    (action) => action.effectId === 'caption'
+                        ? <TimelineCaptionClip action={action} />
+                        : <TimelineVideoClip action={action} />
+                }
+                onClickTimeArea={(time => {
+                    onSetCurrentTime(time)
+                    return true
+                })}
+                onCursorDrag={onSetCurrentTime}
+                onActionMoving={onCueMove}
+                onActionResizing={onCueMove}
+                onClickAction={onCueClick}
+                onChange={setTimelineData}
+                style={{
+                    width: 'unset',
+                    height: '100%',
+                }}
+            />
+
+        </div>
+    )
+}
```

Comment:
This file is getting quite large.

I suggest that you extract some parts (such as zoom control) to separate components before adding more code.

For now it's fine to leave it as is.